src/s3/util/fileViewerManager.ts

```diff
@@ -55,13 +65,32 @@ export class S3FileViewerManager {
         }
         getLogger().verbose(`S3FileViewer: File from s3 or temp to be opened is: ${fileLocation}`)
 
-        //TODOD:: delegate this logic to S3Tab.ts
-        //this will display the document at the end
-        this.window.showTextDocument(fileLocation)
+        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation)
+        await newTab.openFileOnReadOnly()
+
+        this.activeTabs.set(fileLocation.fsPath, newTab)
+    }
+
+    public async openOnEditMode(uriOrNode: vscode.Uri | S3FileNode): Promise<void> {
+        if (uriOrNode instanceof vscode.Uri) {
+            //was activated from an open tab
+            if (this.activeTabs.has(uriOrNode.fsPath)) {
+                const tab = this.activeTabs.get(uriOrNode.fsPath)
+                await tab!.openFileOnEditMode()
+            }
+        } else if (uriOrNode instanceof S3FileNode) {
```

Comment:
Nit: just use `else` (unless there's an error case you should be handling in the `else`)